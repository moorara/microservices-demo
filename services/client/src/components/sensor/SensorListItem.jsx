import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

// Presentational Component
const SensorListItem = ({ sensor }) => (
  <tr>
    <td><Link to={`/sensors/${sensor.id}`}>{sensor.name}</Link></td>
    <td>{sensor.unit}</td>
    <td>{sensor.minSafe}</td>
    <td>{sensor.maxSafe}</td>
  </tr>
)

SensorListItem.propTypes = {
  sensor: PropTypes.object.isRequired
}

export default SensorListItem
