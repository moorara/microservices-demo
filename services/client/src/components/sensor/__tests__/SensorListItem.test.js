import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import SensorListItem from '../SensorListItem'

describe('SensorListItem', () => {
  const props = {
    sensor: {}
  }

  test('renders react component with props', () => {
    props.sensor = { id: 'aaaa-aaaa', siteId: '1111-1111', name: 'temperature', unit: 'celsius', minSafe: -30, maxSafe: 30 }
    const wrapper = shallow(<SensorListItem {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find(`Link [to="/sensors/${props.sensor.id}"]`))
    expect(wrapper.find('td').at(1).text()).toBe(`${props.sensor.unit}`)
    expect(wrapper.find('td').at(2).text()).toBe(`${props.sensor.minSafe}`)
    expect(wrapper.find('td').at(3).text()).toBe(`${props.sensor.maxSafe}`)
  })
  test('renders react component with props', () => {
    props.sesnor = { id: 'bbbb-bbbb', siteId: '2222-2222', name: 'temperature', unit: 'fahrenheit', minSafe: -22, maxSafe: 86 }
    const wrapper = shallow(<SensorListItem {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find(`Link [to="/sensors/${props.sensor.id}"]`))
    expect(wrapper.find('td').at(1).text()).toBe(`${props.sensor.unit}`)
    expect(wrapper.find('td').at(2).text()).toBe(`${props.sensor.minSafe}`)
    expect(wrapper.find('td').at(3).text()).toBe(`${props.sensor.maxSafe}`)
  })
})
