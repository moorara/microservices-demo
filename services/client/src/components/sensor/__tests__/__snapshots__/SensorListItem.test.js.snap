// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensorListItem renders react component with props 1`] = `
<tr>
  <td>
    <Link
      to="/sensors/aaaa-aaaa"
    >
      temperature
    </Link>
  </td>
  <td>
    celsius
  </td>
  <td>
    -30
  </td>
  <td>
    30
  </td>
</tr>
`;

exports[`SensorListItem renders react component with props 2`] = `
<tr>
  <td>
    <Link
      to="/sensors/aaaa-aaaa"
    >
      temperature
    </Link>
  </td>
  <td>
    celsius
  </td>
  <td>
    -30
  </td>
  <td>
    30
  </td>
</tr>
`;
